doc: Static Analysis requirement

ceolin opened the pull request:
Sets static analysis  an indispensable requirement for our project releases.

Static analysis is not merely a tool but a proactive strategy to unearth and address potential issues in the early stages of development, long before they mature into critical vulnerabilities. By scrutinizing code at rest, static analysis unveils latent defects and potential security risks, thus bolstering the resilience of our software against future threats.

Fixes: #64591

Commit:
doc: Static Analysis requirement

Sets static analysis  an indispensable requirement for our project
releases.

Static analysis is not merely a tool but a proactive
strategy to unearth and address potential issues in the early stages
of development, long before they mature into critical
vulnerabilities. By scrutinizing code at rest, static analysis unveils
latent defects and potential security risks, thus bolstering the
resilience of our software against future threats.

Fixes: #64591

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

aasinclair commented on doc/contribute/guidelines.rst:
```diff
@@ -532,9 +532,17 @@ results you have to create an account yourself.  From the Zephyr project page,
+the respective code owner who is responsible for the affected code.
+
+A dedicated team comprising members with expertise in static analysis, code
+quality, and software security is responsible to ensure the effectiveness of
+the static analysis process and verify that identified issues are properly
```
nit: grammar.
I suggest either:
```suggestion
quality, and software security ensure the effectiveness of
the static analysis process and verify that identified issues are properly
```
or
```suggestion
quality, and software security is responsible for ensuring the effectiveness of
the static analysis process and verifying that identified issues are properly
```

keith-zephyr replied:
I like option 1.

kartben replied:
Option 1 would be "ensures", not "ensure"

aasinclair commented on doc/project/project_roles.rst:
```diff
@@ -123,6 +123,9 @@ in addition to those listed for Contributors and Collaborators:
 * Responsibility to ensure all contributions of the project have been reviewed
   within reasonable time.
 * Responsibility to enforce the code of conduct.
+* Responsability to triage static analysis issues in their code area. Improper
```
nit: spelling

aasinclair commented on doc/project/release_process.rst:
```diff
@@ -94,8 +94,14 @@ At that point, the whole process starts over again.
 
-The current backlog of prioritized bugs shall be used as a quality metric to
-gate the final release. The following counts shall be used:
+Static Analysis results a fundamental requirement for the project. The final
```
```suggestion
Static Analysis results are a fundamental requirement for the project. The final
```

aasinclair reviewed (commented):
Looks good, just a couple of grammar/spelling nits.

ceolin commented:
@aasinclair @kartben would you mind take another look ?

str4t0m commented on doc/project/release_process.rst:
```diff
@@ -94,8 +94,15 @@ At that point, the whole process starts over again.
 
-The current backlog of prioritized bugs shall be used as a quality metric to
-gate the final release. The following counts shall be used:
+Static Analysis results are a fundamental requirement for the
+project. The final release must not contain any high-critical issues
+that can potentially compromise the functionality, security, or
+reliability of our software.  High-critical issues represent
+vulnerabilities that, if left unresolved, could have severe
+consequences.
```
Do I understand correctly that this high-critical issues are marked with the "priority: high" Tag? Or is this orthogonal to the low/medium/high priority tagging?

ceolin replied:
yep, you understood correctly. high critical issues are tagged with high priority

aescolar commented on doc/contribute/guidelines.rst:
```diff
@@ -532,9 +532,17 @@ results you have to create an account yourself.  From the Zephyr project page,
+have the same (or equivalent) priority initially defined by the tool.
+
+To ensure accountability and efficient issue resolution, they are assigned to
+the respective code owner who is responsible for the affected code.
```
Nit: I think "maintainer" was better than "codeowner" here.

keith-zephyr replied:
+1

aescolar commented on doc/project/project_roles.rst:
```diff
@@ -123,6 +123,9 @@ in addition to those listed for Contributors and Collaborators:
   within reasonable time.
 * Responsibility to enforce the code of conduct.
+* Responsability to triage static analysis issues in their code area. Improper
+  triage may lead hidden defects and potential security risks.
```
nit: "lead" sounds weird. maybe "result" would be better? 
In any case, this would be the only point with a listed motivation, so maybe that whole sentence does not belong here?

keith-zephyr replied:
I agree with dropping the sentence. Alternatively, you could add this motivation to the static analysis audit team.

aescolar commented on doc/project/release_process.rst:
```diff
@@ -94,8 +94,15 @@ At that point, the whole process starts over again.
 
-The current backlog of prioritized bugs shall be used as a quality metric to
-gate the final release. The following counts shall be used:
+Static Analysis results are a fundamental requirement for the
```
I would not place this paragraph first, but after the current one and the bug count table, maybe as a note.

The first sentence "Static Analysis results are a fundamental requirement for the project." seems a bit off topic in the "Release Quality Criteria" section, and more in place in the "Static Code Analysis" section.

aescolar commented on doc/project/project_roles.rst:
```diff
@@ -170,6 +173,26 @@ the latter is not possible.
 
 .. _release-engineering-team:
 
+Static Analysis Audit Team
```
The _release-engineering-team tag should be next to its section.
(maybe this new section should just be below the release engineering team section)

ceolin replied:
good catch !

aescolar reviewed (commented):
A few minor things

keith-zephyr commented on doc/contribute/guidelines.rst:
```diff
@@ -532,9 +532,17 @@ results you have to create an account yourself.  From the Zephyr project page,
-created for any problems found and assigned to the maintainers of the affected
-areas.
+Static analysis in the Zephyr codebase happens bi-weekly. GitHub issues are
+automatically created for any issues detected by Coverity. These issues will
```
```suggestion
automatically created for any issues detected by the static analysis tool. These issues will
```

kartben commented on doc/project/release_process.rst:
```diff
@@ -269,8 +276,9 @@ components provided by the project:
   conventions and documentation of deviations.
-- Regular static analysis on the complete tree using available commercial and
-  open-source tools and documentation of deviations and false positives.
+- Static analysis report
```
Since the document is leaving the door open to having multiple code analysis tools (just look at line 280 below, for example), I was under the assumption that there might be several report*S* potentially being made available?

ceolin replied:
ok, fair enough.

ceolin commented:
Just noticed that I had pushed the fixes to a wrong branch. @kartben @aescolar could you take  a look on it again please ?

kartben commented on doc/contribute/guidelines.rst:
```diff
@@ -532,9 +532,18 @@ results you have to create an account yourself.  From the Zephyr project page,
-Coverity scans the Zephyr codebase weekly. GitHub issues are automatically
-created for any problems found and assigned to the maintainers of the affected
-areas.
+Static analysis of the Zephyr codebase is conducted on a bi-weekly. GitHub
```
previous request for change was only partially incorporated leading to a missing word:

```suggestion
Static analysis of the Zephyr codebase is conducted on a bi-weekly basis. GitHub
```

kartben commented on doc/contribute/guidelines.rst:
```diff
@@ -532,9 +532,18 @@ results you have to create an account yourself.  From the Zephyr project page,
+
+A dedicated team comprising members with expertise in static analysis, code
+quality, and software security ensures the effectiveness of the static
+analysis process and verify that identified issues are properly
```
```suggestion
analysis process and verifies that identified issues are properly
```

kartben commented on doc/project/project_roles.rst:
```diff
@@ -168,6 +170,26 @@ the latter is not possible.
+cycle are properly addressed. The team has the following rights and
+responsibilities:
+
+* Right to revert any changes in a static analysis tool (e.g: Coverity)
```
```suggestion
* Right to revert any change in a static analysis tool (e.g: Coverity)
```

kartben commented on doc/project/release_process.rst:
```diff
@@ -109,6 +109,11 @@ gate the final release. The following counts shall be used:
    The "low" bug count target of <50 will be a phased approach starting with 150
    for release 2.4.0, 100 for release 2.5.0, and 50 for release 2.6.0
 
+The final release must not contain any static analysis high-critical
+issues that can potentially compromise the functionality, security, or
+reliability of our software.  High-critical issues represent
+vulnerabilities that, if left unresolved, could have severe
+consequences.
```
would recommend you check your formatter, it looks like it wraps at 72 characters which seems overly aggressive -- 100 would be better. Arguably, the file is currently pretty inconsistently formatted with wrapping a bit all over the place.

kartben reviewed (changes requested):
spotted a previous request for change that wasn't fully addressed and add another quick pass leading to uncovering a couple additional grammar mistakes.

Would be nice to also properly reformat the file to use 100-character line wrapping, or at a minimum make sure that your changes are not wrapping at 72 characters like it seems to be the case in several occurences.

Thanks!

ceolin commented:
@kartben thanks for reviewing it again :) I may have missed some of these changes when pushing it to a wrong branch :/

Have addresses these comments now, please check it again when you can.
